File: compiler/bench/simd4.c
// expected: 3334
// Count matching elements - vectorizable conditional count
#include <stdio.h>
#define SIZE 10000
long arr[SIZE];

int main() {
    for (int i = 0; i < SIZE; i++) arr[i] = i % 3;
    long count = 0;
    for (int i = 0; i < SIZE; i++) if (arr[i] == 0) count++;
    printf("%ld\n", count);
    return 0;
}

File: compiler/bench/simd4.fs
\ expected: 3334
\ Count matching elements - vectorizable conditional count

create arr 80000 allot

: init ( -- )
  10000 0 do
    i 3 mod i 8 * arr + !
  loop ;

: count-zeros ( -- n )
  0 10000 0 do
    i 8 * arr + @ 0= if 1+ then
  loop ;

: main init count-zeros . cr ;
